File: 2265-count-nodes-equal-to-average-of-subtree/2265-count-nodes-equal-to-average-of-subtree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
   
    def averageOfSubtree(self, root: TreeNode) -> int:
        self.count=0
        self.postOrder(root)
        return self.count
    
    def postOrder(self,root):
        if not root:
            return 0,0
        left_sum,left_count=self.postOrder(root.left)
        right_sum,right_count=self.postOrder(root.right)
        cur_sum=left_sum+right_sum+root.val
        cur_count=left_count+right_count+1
        if cur_sum // cur_count==root.val:
            self.count+=1
        return cur_sum,cur_count

File: 2265-count-nodes-equal-to-average-of-subtree/2265-count-nodes-equal-to-average-of-subtree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    int total_count=0;
    public int averageOfSubtree(TreeNode root) {
        dfs(root);
        return total_count;
    }
    public int[] dfs(TreeNode root){
        if(root==null)
            return new int[]{0,0};
        int left[]=dfs(root.left);
        int right[]=dfs(root.right);
        int total=left[0]+right[0]+root.val;
        int count=left[1]+right[1]+1;
        if(total/count == root.val)
            total_count++;
        return new int[]{total,count};
    }
}